### ISSUE TITLE
`structonly` only works when combined with `required`

### ISSUE THREAD

**Issue by thessem** [go-playground/validator#264]
### Package version eg. v8, v9: 

"gopkg.in/go-playground/validator.v9"

### Issue, Question or Enhancement:

The `structonly` tag seems to do nothing when it isn't combined with the tag required

### Code sample, to showcase or reproduce:

```go
package main

import (
	"log"

	"gopkg.in/go-playground/validator.v9"
)

type User struct {
	FirstName      string     `json:"fname"`
	LastName       string     `json:"lname"`
	Age            uint8      `validate:"gte=0,lte=130"`
	Email          string     `validate:"required,email"`
	FavouriteColor string     `validate:"hexcolor|rgb|rgba"`
	Addresses      []*Address `validate:"required"`            // a person can have a home and cottage...
	Address        Address    `validate:"required,structonly"` // a person can have a home and cottage...
}

// Address houses a users address information
type Address struct {
	Street string `validate:"required"`
	City   string `validate:"required"`
	Planet string `validate:"required"`
	Phone  string `validate:"required"`
}

// use a single instance of Validate, it caches struct info
var validate *validator.Validate

func main() {
	validate = validator.New()
	address := &Address{
		Street: "Eavesdown Docks",
		Planet: "Persphone",
		Phone:  "none",
		City:   "Unknown",
	}

	user := &User{
		FirstName:      "",
		LastName:       "",
		Age:            45,
		Email:          "Badger.Smith@gmail.com",
		FavouriteColor: "#000",
		Addresses:      []*Address{address},
		Address: Address{
			// Street: "Eavesdown Docks",
			Planet: "Persphone",
			Phone:  "none",
			City:   "Unknown",
		},
	}
	err := validate.Struct(user)
	if err != nil {
		log.Println(err)
	}
}
```

if the line above is changed from
```go
Address        Address    `validate:"required,structonly"` // a person can have a home and cottage...
```
to
```go
Address        Address    `validate:"structonly"` // a person can have a home and cottage...
```
the error
```
Key: 'User.Address.Street' Error:Field validation for 'Street' failed on the 'required' tag
```
is given, but not if it is left as `required,structonly`.

Additionally, is there any way to turn off nested struct validation, e.g., make `structonly` the default behaviour?

**Comment by joeybloggs** [comment]
Thanks for reporting @thessem it ha been corrected in [release v9.3.1](https://github.com/go-playground/validator/releases/tag/v9.3.1)

Please test yourself and let me know that it's working for you as well 😄 

As for making `structonly` the default behaviour currently there is no way to set that as the default behaviour thanks you for the idea though, I'll add this to a list of future additions much like it works in [go-playground/form](https://github.com/go-playground/form/blob/master/form.go#L27)

This is just an assumption so please correct me if this is not correct, I assume if you're looking for `structonly` as the default behaviour then your using `structlevel` validation only, if that is the case and you don't have any validations on the struct fields themselves then they won't get cached and fields will not be checked every time if worried about performance.

**Comment by thessem** [comment]
I was using `structonly` to avoid recursing into the nested struct's fields.

My structs are a graph connected to each other by pointers where nodes can have properties (fields) and links to other nodes (represented by a pointer to another struct). I was hoping to avoid recursing through the structs because I already have a flat list of structs in an `[]interface{}`.

This was a bit of a hack, but I figured if I appended every field with `structonly` while validating every struct in the array I would eventually fully validate my memory structure.

Thank you for the extremely fast response!
